sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java

```diff
@@ -726,7 +746,16 @@ public String[] fieldNames() {
      * is not provided as updating column default values does not need to back-fill existing data.
      * Empty string means dropping the column default value.
      */
-    public String newDefaultValue() { return newDefaultValue; }
+    public String newDefaultValue() {
+      return newDefaultValue == null ? "" : newDefaultValue.getSql();
+    }
+
+    /**
+     * Returns the column default value as {@link DefaultValue}. The default value literal
+     * is not provided as updating column default values does not need to back-fill existing data.
+     * Empty string and Null literal means dropping the column default value.
+     */
+    public DefaultValue newModelDefaultValue() { return newDefaultValue; }
 
     @Override
     public boolean equals(Object o) {
```

Comment:
We have to update `equals` and `hashCode` now too.

Reply:
done